xarray/conventions.py

```diff
@@ -165,11 +165,18 @@ def decode_cf_timedelta(num_timedeltas, units):
     """
     num_timedeltas = _asarray_or_scalar(num_timedeltas)
     units = _netcdf_to_numpy_timeunit(units)
+
+    shape = None
+    if isinstance(num_timedeltas, np.ndarray):
```

Comment:
I'm not 100% sure, but I think the `_asarray_or_scalar` is there because pandas's `to_timedelta` doesn't like receiving 0-dimensional numpy arrays as input. As long as you're flattening everything you pass to `to_timedelta` I think arrays should always be OK. Tests should tell us either way.